iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/writer/WriterBuilder.java

```diff
@@ -92,7 +92,10 @@ public HiveIcebergWriter build() {
     long targetFileSize = PropertyUtil.propertyAsLong(table.properties(), TableProperties.WRITE_TARGET_FILE_SIZE_BYTES,
         TableProperties.WRITE_TARGET_FILE_SIZE_BYTES_DEFAULT);
 
+    boolean isWriteDeleteRow = Boolean.parseBoolean(properties.getOrDefault("iceberg.write.deleterow", "false"));
+
     Schema dataSchema = table.schema();
+    Schema positionalDeleteSchema = isWriteDeleteRow ? dataSchema : null;
```

Comment:
not sure if the naming is good, it's actually a data schema. Maybe better to just inline this ternary operator in Factory constructor?